pkg/globalstats/stats.go

```diff
@@ -125,7 +125,19 @@ func Initialize(ctx context.Context, config Config) {
 
 func flusher(ctx context.Context, stop <-chan struct{}, flushEvery time.Duration) {
 	defer engine.Flush()
+
 	for {
+		// segmentio/stats defaults to discarding metrics. If we initialize globalstats from ctlstore.init,
+		// then the user will not yet have been able to configure their own handler. In that case, we need
+		// to check if the user has supplied a handler, in which case we swap for that one.
+		if engine.Handler == stats.Discard {
+			if stats.DefaultEngine.Handler != stats.Discard {
+				engine = stats.NewEngine(engine.Prefix, stats.DefaultEngine.Handler, engine.Tags...)
```

Comment:
This seems like a race between this goroutine that is setting engine and the other goroutines that access the same variable through the globalstats API.

Reply:
Fixed!